packages/server/lib/gui/links.ts

```diff
@@ -1,28 +1,36 @@
 import _ from 'lodash'
 import { shell } from 'electron'
+import { machineId } from '../util/machine_id'
 
 // NOTE: in order for query params to be passed through on links
 // forwardQueryParams: true must be set for that slug in the on package
 
 interface OpenExternalOptions {
   url: string
-  params: { [key: string]: string }
+  params?: { [key: string]: string }
+  machineId?: boolean
 }
 
-export const openExternal = (opts: OpenExternalOptions | string) => {
+export const openExternal = async (opts: OpenExternalOptions | string) => {
   if (_.isString(opts)) {
     return shell.openExternal(opts)
   }
 
   const url = new URL(opts.url)
 
-  if (opts.params) {
+  const params = opts.params || {}
+
+  if (opts.machineId && url.hostname === 'on.cypress.io') {
+    params.machine_id = await machineId()
+  }
```

Comment:
there needs to be a way to opt out of this, for example we use the `CYPRESS_CRASH_REPORTS` environment variable as an opt out here: https://github.com/cypress-io/cypress/blob/3726d7eb26f6489ef6ceb8c046de0f239f830e81/packages/server/lib/exception.js#L54-L64